addons/medical_engine/functions/fnc_getHitpointArmor.sqf

```diff
@@ -0,0 +1,78 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Checks a unit's equipment to calculate the total armor on a hitpoint
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Hitpoint <STRING>
+ *
+ * Return Value:
+ * Total armor for the given hitpoint <ARRAY>
+ * 0: Armor <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * [player, "HitChest"] call ace_medical_engine_fnc_getHitpointArmor
+ *
+ * Public: No
+ */
+params ["_unit", "_hitpoint"];
+
+// Helper function to check items with HitpointsProtectionInfo
+private _fnc_getItemArmor = {
+    params ["_item", "_hitpoint"];
+    if (_item isEqualTo "") exitWith {[0,0]};
+    private _key = format ["%1$%2", _item, _hitpoint];
+    private _armorValues = GVAR(armorCache) getVariable _key;
+    if (isNil "_armorValues") then {
+        TRACE_2("Cache miss", _item, _hitpoint);
+        private _cfg = configFile >> "CfgWeapons" >> _item >> "ItemInfo" >> "HitpointsProtectionInfo";
+        private _condition = format ["getText (_x >> ""hitpointName"") == ""%1""",_hitpoint];
+        private _entry = configProperties [_cfg, _condition] param [0, configNull];
+        _armorValues = [getNumber (_entry >> "armor"), getNumber (_entry >> "explosionShielding")];
+        GVAR(armorCache) setVariable [_key, _armorValues];
+    };
+    _armorValues // return
+};
+
+private _uniform = uniform _unit;
+// If unit is naked, use its underwear class instead
+if (_uniform isEqualTo "") then {
+    _uniform =  getText (configFile >> "CfgVehicles" >> typeOf _unit >> "nakedUniform");
+};
+
+// Uniform base values are a multiplier for all hitpoint values
+private _baseValues = GVAR(armorCache) getVariable (_uniform + "$armor");
+if (isNil "_baseValues") then {
+    TRACE_1("Cache miss", _uniform);
+    private _uniformClass = getText (configFile >> "CfgWeapons" >> _uniform >> "ItemInfo" >> "uniformClass");
+    private _unitCfg = configFile >> "CfgVehicles" >> _uniformClass;
```

Comment:
Considering this is performance relevant, you maybe shouldn't re-lookup CfgWeapons/CfgVehicles everytime, and instead do it once and store it in a local.
`_fnc_getItemArmor`  could also use that local then.

Reply:
but also considering that stuff is cached, that might cause two useless lookups everytime..

Reply:
As per commy's points, we expect the majority of calls to hit the cache so it's better to leave as much as possible until after that. A couple of extra `>>` only on cache miss is better than wasted processing on every event.
It _could_ be declared in the parent scope and only actually set/used if we do a lookup; but that means an extra `isNil` check, and is less readable.